python/pyarrow/tests/test_dataset.py

```diff
@@ -760,6 +760,98 @@ def test_fragments_parquet_row_groups(tempdir):
     assert len(result) == 1
 
 
+def _create_dataset_all_types(tempdir, chunk_size=None):
+    import pyarrow.parquet as pq
+
+    table = pa.table(
+        [
+            pa.array([True, None, False], pa.bool_()),
+            pa.array([1, 10, 42], pa.int8()),
+            pa.array([1, 10, 42], pa.uint8()),
+            pa.array([1, 10, 42], pa.int16()),
+            pa.array([1, 10, 42], pa.uint16()),
+            pa.array([1, 10, 42], pa.int32()),
+            pa.array([1, 10, 42], pa.uint32()),
+            pa.array([1, 10, 42], pa.int64()),
+            pa.array([1, 10, 42], pa.uint64()),
+            pa.array([1.0, 10.0, 42.0], pa.float32()),
+            pa.array([1.0, 10.0, 42.0], pa.float64()),
+            pa.array(['a', None, 'z'], pa.utf8()),
+            # pa.array(['a', None, 'z'], pa.binary()),
```

Comment:
That's to be enabled once #7519 is merged, e.g. it is missing python Scalar classes.

Reply:
Note that it should work since I rebased this branch on top of #7519 and uncommented.